integration/helpers/client_provider.py

```diff
@@ -75,3 +80,48 @@ def sfn_client(self):
         if not self._sfn_client:
             self._sfn_client = boto3.client("stepfunctions")
         return self._sfn_client
+
+    @property
+    def cloudwatch_log_client(self):
+        """
+        CloudWatch Log Client
+        """
+        if not self._cloudwatch_log_client:
```

Comment:
As we will run these tests in parallel in the future, we may want to make this class thread safe.

Import
`from threading import Lock`

Add the class attribute in init:
`self._lock = Lock()`

Use as follows:
```
if not self._cloudwatch_log_client:
    with self._lock:
        if not self._cloudwatch_log_client:
            self._cloudwatch_log_client = boto3.client("logs")
return self._cloudwatch_event_client
```

Reply:
Good idea, will implement it.